test/topology_experimental_raft/test_fencing.py

```diff
@@ -63,3 +99,78 @@ async def test_fence_writes(request, manager: ManagerClient):
         await manager.cql.run_async("update t2 set counter_c=counter_c+1 where pk=1", host=host2)
 
     random_tables.drop_all()
+
+
+@pytest.mark.asyncio
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_fence_hints(request, manager: ManagerClient):
+    logger.info("Bootstrapping cluster with three nodes")
+    s0 = await manager.server_add(config={
+        'error_injections_at_startup': ['decrease_hints_flush_period']
+    })
+    s1 = await manager.server_add()
+    s2 = await manager.server_add()
+
+    logger.info(f'Creating test table')
+    random_tables = RandomTables(request.node.name, manager, unique_name(), 3)
+    table1 = await random_tables.add_table(name='t1', pks=1, columns=[
+        Column("pk", IntType),
+        Column('int_c', IntType)
+    ])
+    await manager.cql.run_async(f"USE {random_tables.keyspace}")
+
+    logger.info(f'Waiting for cql and hosts')
+    hosts = await wait_for_cql_and_get_hosts(manager.cql, [s0, s2], time.time() + 60)
+
+    host2 = host_by_server(hosts, s2)
+    new_version = (await get_version(manager, host2)) + 1
+    logger.info(f"Set version and fence_version to {new_version} on node {host2}")
+    await set_version(manager, host2, new_version)
+    await set_fence_version(manager, host2, new_version)
+
+    select_all_stmt = SimpleStatement("select * from t1", consistency_level=ConsistencyLevel.ONE)
+    rows = await manager.cql.run_async(select_all_stmt, host=host2)
+    assert len(list(rows)) == 0
+
+    logger.info(f"Stopping node {host2}")
+    await manager.server_stop_gracefully(s2.server_id)
+
+    host0 = host_by_server(hosts, s0)
+    logger.info(f"Writing through {host0} to regular column")
+    await manager.cql.run_async("insert into t1(pk, int_c) values (1, 1)", host=host0)
+
+    logger.info(f"Starting last node {host2}")
+    await manager.server_start(s2.server_id)
+
+    logger.info(f"Waiting for failed hints on {host0}")
+    async def at_least_one_hint_failed():
+        metrics_data = await manager.metrics.query(s0.ip_addr)
+        if send_errors_metric(metrics_data) >= 1 and sent_metric(metrics_data) == 0:
+            return True
+        logger.info(f"Metrics on {s0}: {metrics_data.lines_by_prefix('scylla_hints_manager_')}")
+    await wait_for(at_least_one_hint_failed, time.time() + 5)
+
+    host2 = (await wait_for_cql_and_get_hosts(manager.cql, [s2], time.time() + 60))[0]
+
+    # Check there is no new data on host2.
+    rows = await manager.cql.run_async(select_all_stmt, host=host2)
```

Comment:
Is probabilistic read repair disabled for the random table? If it is enabled then this select might accidentally repair the data without using hints.